src/Microsoft.Diagnostics.DebugServices.Implementation/Runtime.cs

```diff
@@ -131,62 +121,45 @@ private ClrRuntime CreateRuntime()
             return _clrRuntime;
         }
 
-        private string GetDacFileName()
+        private string GetLibraryPath(DebugLibraryKind kind)
         {
-            if (_clrInfo.SingleFileRuntimeInfo.HasValue)
-            {
-                return ClrInfoProvider.GetDacFileName(_clrInfo.Flavor, Target.OperatingSystem);
-            }
-            Debug.Assert(!string.IsNullOrEmpty(_clrInfo.DacInfo.PlatformSpecificFileName));
-            return _clrInfo.DacInfo.PlatformSpecificFileName;
-        }
+            Architecture currentArch = RuntimeInformation.ProcessArchitecture;
+            string libraryPath = null;
 
-        private string GetLocalDacPath(string dacFileName)
-        {
-            string dacFilePath;
-            if (!string.IsNullOrEmpty(RuntimeModuleDirectory))
-            {
-                dacFilePath = Path.Combine(RuntimeModuleDirectory, dacFileName);
-            }
-            else
+            foreach (DebugLibraryInfo libraryInfo in _clrInfo.DebuggingLibraries)
             {
-                dacFilePath = _clrInfo.DacInfo.LocalDacPath;
-
-                // On MacOS CLRMD doesn't return the full DAC path just the file name so check if it exists
-                if (string.IsNullOrEmpty(dacFilePath) || !File.Exists(dacFilePath))
+                if (libraryInfo.Kind == kind && RuntimeInformation.IsOSPlatform(libraryInfo.Platform) && libraryInfo.TargetArchitecture == currentArch)
                 {
-                    dacFilePath = Path.Combine(Path.GetDirectoryName(RuntimeModule.FileName), dacFileName);
+                    libraryPath = GetLocalPath(libraryInfo.FileName);
+                    if (libraryPath is not null)
+                    {
+                        break;
+                    }
+                    libraryPath = DownloadFile(libraryInfo);
+                    if (libraryPath is not null)
+                    {
+                        break;
+                    }
                 }
             }
-            if (!File.Exists(dacFilePath))
-            {
-                dacFilePath = null;
-            }
-            return dacFilePath;
-        }
-
-        private string GetDbiFileName()
-        {
-            string name = Target.GetPlatformModuleName("mscordbi");
 
-            // If this is the Linux runtime module name, but we are running on Windows return the cross-OS DBI name.
-            if (Target.OperatingSystem == OSPlatform.Linux && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                name = "mscordbi.dll";
-            }
-            return name;
+            return libraryPath;
         }
 
         private string GetLocalPath(string fileName)
         {
+            if (File.Exists(fileName))
```

Comment:
You could run into the issue where there is a stray "mscordaccore.dll" laying in the current working directory and it would load that if fileName isn't an absolute path.  I would use `if (fileName != Path.GetFileName(fileName) && File.Exists(fileName))` instead.

Maybe I should change this in ClrMD...but I'll leave that for next minor version bump.